frontend/src/pages/Rubrics/RubricEditor/index.tsx

```diff
@@ -0,0 +1,215 @@
+import Button from "@/components/Button";
+import Input from "@/components/Input";
+import RubricItem from "@/components/RubricItem";
+import { createRubric, updateRubric } from "@/api/rubrics";
+import { SelectedClassroomContext } from "@/contexts/selectedClassroom";
+import { setAssignmentRubric } from "@/api/assignments";
+import { useLocation, useNavigate } from "react-router-dom";
+
+interface IEditableItem {
+    frontFacingIndex: number;
+    rubricItem: IRubricItem;
+}
+
+
+const RubricEditor: React.FC = () => {
+    const location = useLocation()
+    const navigate = useNavigate();
+
+    const { selectedClassroom } = useContext(SelectedClassroomContext)
+
+    // potential data for the assignment
+    const [assignmentData, setAssignmentData] = useState<IAssignmentOutline>()
+    //data for the rubric
+    const [rubricData, setRubricData] = useState<IFullRubric>()
+    const [rubricItemData, setRubricItemData] = useState<IEditableItem[]>([])
+    const [rubricName, setRubricName] = useState<string>("")
+
+    // if there has been any changes since last save
+    const [rubricEdited, setRubricEdited] = useState<boolean>(false)
+
+    // front end id for each rubric item, kept track of using a counter
+    const [itemCount, setitemCount] = useState(0);
+    const incrementCount = () => setitemCount(itemCount + 1);
+
+    // default item for adding new items
+    const newRubricItem: IEditableItem = {
+        frontFacingIndex: itemCount,
+        rubricItem: {
+            id: null,
+            point_value: 0,
+            explanation: "",
+            rubric_id: null,
+            created_at: null
+        }
+    }
+
+    const backButton = () => {
+        navigate(-1);
+    }
+    // saving the rubric, creates a rubric in the backend
+    const saveRubric = async () => {
+        if (selectedClassroom !== null && selectedClassroom !== undefined && rubricEdited) {
+            const rubricItems = (rubricItemData.map(item => item.rubricItem));
+
+            const fullRubric: IFullRubric = {
+                rubric: {
+                    id: null,
+                    name: rubricName,
+                    org_id: selectedClassroom.org_id,
+                    classroom_id: selectedClassroom.id,
+                    reusable: true,
+                    created_at: null
+                },
+                rubric_items: rubricItems
+            }
+
+            // update existing rubric
+            if (rubricData) {
+                await updateRubric(rubricData.rubric.id!, fullRubric)
+                    .then((updatedRubric) => {
+                        setRubricEdited(false)
+                        setRubricData(updatedRubric)
+                        if (assignmentData !== null && assignmentData !== undefined) {
+                            setAssignmentRubric(updatedRubric.rubric.id!, selectedClassroom.id, assignmentData.id)
+                        }
+                    })
+                    .catch((_) => {
```

Comment:
we should have error states. Also, perhaps a loading state as well would be awesome if there isn't one already